buckets/local/bucket.go

```diff
@@ -0,0 +1,297 @@
+package local
```

Comment:
This package has a large part of the new work. The general idea (covered in the video), is to maintain a local car file that represents the remote bucket, which is used to compare changed in the working directory.